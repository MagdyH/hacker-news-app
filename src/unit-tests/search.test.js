import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../components/search';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

const initialState = {}
const mockStore = configureStore();
const store = mockStore(initialState);


describe('Search component', () => {
    it('should render snapshot', () => {
        const component = renderer.create(<Search store={store} />);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});

describe('Search component', () => {
    it('should call onChange', () => {
        const component = mount(<Search store={store} />);
        const event = {
            preventDefault() {},
            target: { value: 'test' }
          };
          const input  = component.find('input');
          
          input.simulate('change', event);
          expect(event.target.value).toEqual('test');
    });

});