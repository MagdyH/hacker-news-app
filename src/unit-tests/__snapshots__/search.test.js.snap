// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render snapshot 1`] = `
<div
  className="Search-container"
>
  <span
    className="Search-Icon"
  >
    <i
      aria-hidden="true"
      className="fa fa-search"
    />
  </span>
  <input
    aria-label="Search"
    className="Search-Input"
    onChange={[Function]}
    placeholder="Search stories by title, url or author"
    type="text"
  />
</div>
`;

exports[`Search component should render snapshot 2`] = `
<div
  className="Search-container"
>
  <span
    className="Search-Icon"
  >
    <i
      aria-hidden="true"
      className="fa fa-search"
    />
  </span>
  <input
    aria-label="Search"
    className="Search-Input"
    onChange={[Function]}
    placeholder="Search stories by title, url or author"
    type="text"
  />
</div>
`;
